src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int START_VALUE = 0;
+    private static final int STEP = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyValue = calculateSupplyValue(readFromFile(fromFileName));
+        int buyValue = calculateBuyValue(readFromFile(fromFileName));
+        writeInfoToFile(supplyValue, buyValue, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+                new FileReader(new File(fromFileName)))) {
+            StringBuilder builder = new StringBuilder();
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                String[] listArray = line.split(",");
```

Comment:
Instead of using 'split()' inside the loop and then concatenating the results, you can read each line and append it to the StringBuilder with a delimiter, then split the final string after the loop.